internal/pkg/template/templatetest/templatetest.go

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package templatetest provides test doubles for embedded templates.
```

Comment:
what is "test doubles" 🤔

Reply:
https://testing.googleblog.com/2013/07/testing-on-toilet-know-your-test-doubles.html

It's the general term for giving a "test object" instead of the real one.
In our code, we typically have:
- Stubs: the custom structs that we ourselves write in the `_test.go` files that return some dummy data.
- Mocks: we use the gomock library to test if an object was called with the right input and how many times.
- Fakes: very rarely we use fakes, the best example is `afero.NewMemMapFS`, but they are really nice for high fidely cheap integ tests